unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -14,11 +17,24 @@ static unsafe partial class CoreCLRHost
 {
     static ALCWrapper alcWrapper;
     static FieldInfo assemblyHandleField;
+    private static readonly ConcurrentDictionary<Assembly, AssemblyPair> m_assemblies = new ();
```

Comment:
I would switch back to `Dictionary` now that we are using a lock.  I imagine `ConcurrentDictionary` comes with extra overhead that wouldn't be providing any value.

Reply:
True. All access is behind a lock anyway now.